cpp/src/lists/explode.cu

```diff
@@ -208,38 +208,45 @@ std::unique_ptr<table> explode_outer(table_view const& input_table,
   // offsets + 1 here to skip the 0th offset, which removes a - 1 operation later.
   auto offsets_minus_one = thrust::make_transform_iterator(
     thrust::next(offsets), [offsets] __device__(auto i) { return (i - offsets[0]) - 1; });
+
+  // we need to do this loop at least explode_col times or we may not properly fill in null and
+  // empty entries.
+  auto loop_count = std::max(sliced_child.size(), explode_col.size());
+
   // Fill in gather map with all the child column's entries
   thrust::for_each(rmm::exec_policy(stream),
```

Comment:
Is there any way to break this huge `for_each` into smaller `for_each`?

Reply:
I originally had this as two different `for_each` loops, but as they were over similar sizes, it seemed better to roll them together. That is actually what caused this bug though and I could split them apart again, but I worry about the performance of it even if it is `O(n)` vs `O(2n)` since `n` is expected to be very large. Maybe I'm premature with this concern.